vm/actor/src/builtin/market/mod.rs

```diff
@@ -1251,12 +1251,6 @@ where
     BS: BlockStore,
     RT: Runtime<BS>,
 {
-    if proposal.proposal.end_epoch <= proposal.proposal.start_epoch {
-        return Err(actor_error!(
-            ErrIllegalArgument,
-            "proposal end epoch before start epoch"
-        ));
-    }
```

Comment:
I think this check still needs to happen, but me moved after the `verify_signature` step?

Reply:
mmh the check is already happening [here](https://github.com/ChainSafe/forest/blob/main/vm/actor/src/builtin/market/mod.rs#L1185-L1190).
I don't see the point to do it twice. Lotus does it only once too if I recall.